### ISSUE TITLE
Need to add webdav function

### ISSUE THREAD

**Issue by wanglixiang90** [filebrowser/filebrowser#2130]
**Additional context**
Need to add webdav function .
The webdav function needs to be added with basic authentication. I hope to support ldap authentication

**Comment by stokito** [comment]
This was already asked #1955 #1138 #1038 but let's discuss it in the new ticket. Ldap and auth is slightly different thing and you may find other tickets.

Here is two different things:
1. Backend support: To expose the WebDAV API on the backend so that users can mount the share.
2. Frontend support: to see an existing WebDAV share from frontend. I.e. not use the FileBrowser backend at all.

As far I understood the 1 Backend support is currently developing in the fork https://github.com/browsefile/backend 

@vitaliy-kuzmich could you please confirm? Will you send a PR when finishing?

The Frontend support of the WebDAV is something that is hard to implement simply because the WebDAV API is only about raw files and folders. It doesn't have users management, authorization, thumbnails and other features.
Still even with limited set of features it would be great to have it.
For example I installed Lighttpd + mod_webdav on my old router with USB disk and now I have a very cheap and fast NAS.
But I need for in-browser UI and I used the https://github.com/dom111/webdav-js which is a small JS script that sends WebDAV's `PROPFIND` request to list directory content and renders it. 
Now I wish to have slightly more features like thumbnails and I had an interesting discussion https://github.com/dom111/webdav-js/issues/94

So both authors of the webdav-js and the Lighttpd are also interested in developing some rich file browser based on the WebDAV.

The FileBrowser has impressive UI almost like Google Docs. So what if you can take the webdav-js library to interact with the WebDAV protocol and use it as you now use own protocol?

For example when I click on a file I see a request and response:
```
GET http://localhost:8080/api/resources/Docs/
{
  "items": [
    {
      "path": "/Docs/test.txt",
      "name": "test.txt",
      "size": 1635,
      "extension": ".txt",
      "modified": "2022-09-14T20:57:01.075267682Z",
      "mode": 436,
      "isDir": false,
      "isSymlink": false,
      "type": "text"
    }
}
```

So the response is in JSON format but basically it contains almost the same data as the WebDAV `PROPFIND` response has:
```xml
<?xml version="1.0" encoding="utf-8"?>
<D:multistatus xmlns:D="DAV:">
    <D:response>
        <D:href>/Docs/test.txt</D:href>
        <D:propstat>
            <D:prop>
                <D:getcontentlength>1635</D:getcontentlength>
                <D:getcontenttype>text/plain</D:getcontenttype>
                <D:getlastmodified>Sat, 14 Sep 2022 03:09:43 GMT</D:getlastmodified>
            </D:prop>
            <D:status>HTTP/1.1 200 OK</D:status>
        </D:propstat>
    </D:response>
</D:multistatus>
```

The WebDAV response lacks of `mode` or `isSymlink` but at least basic info can be shown. And yes, that's XML but not a big deal.
The NextCloud API is also based on the WebDAV and they just adding their custom tags.
Basically we can ask author of the Lighttpd to add some config option like "extended tags" and include some additional tags that aren't specified by WebDAV specs.

BTW there is a lot of other JS clients for WebDAV that may be used https://www.npmjs.com/search?q=webdav

**Comment by bkil** [comment]
Are there any other common backend API that should be supported? I.e., if there exist a few other common ones, it might be a good idea to consider creating a storage API abstraction within Filebrowser and then implement WebDAV and other backends as modules or plugins to that.

**Comment by vitaliy-kuzmich** [comment]
> @vitaliy-kuzmich could you please confirm? Will you send a PR when finishing?

browsefile is absolutely different project. I started developing it because nobody wants to listen to me at filebrowser project! My offer was to cancel DB, because it is useless, filesystem is already DB by itself!

**Comment by boniek83** [comment]
Webdav could be used to expose kubernetes volumes over http and use rsync or rclone to ingest data to them with ability to resume if anything goes wrong or just send delta instead of whole dataset. Very useful feature.

**Comment by vitaliy-kuzmich** [comment]
> Webdav could be used to expose kubernetes volumes over http and use rsync or rclone to ingest data to them with ability to resume if anything goes wrong or just send delta instead of whole dataset. Very useful feature.

there is alternative called browsefile it supports:
-webdav include auth
-filesharing between users
-has no db, instead it uses simple json config file

**Comment by boniek83** [comment]
> > Webdav could be used to expose kubernetes volumes over http and use rsync or rclone to ingest data to them with ability to resume if anything goes wrong or just send delta instead of whole dataset. Very useful feature.
> 
> there is alternative called browsefile it supports: -webdav include auth -filesharing between users -has no db, instead it uses simple json config file

Can you provide link to this project?

**Comment by boniek83** [comment]
> > Webdav could be used to expose kubernetes volumes over http and use rsync or rclone to ingest data to them with ability to resume if anything goes wrong or just send delta instead of whole dataset. Very useful feature.
> 
> there is alternative called browsefile it supports: -webdav include auth -filesharing between users -has no db, instead it uses simple json config file

Thanks! Isn't it basically dead though?
